### ISSUE TITLE
TypeError in DoctrineBackendOptimizeListener

### ISSUE THREAD

**Issue by jo66** [sonata-project/SonataNotificationBundle#430]
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.63.0 3.63.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.18.4 3.18.4 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/classification-bundle     3.11.1 3.11.1 Symfony SonataClassificationBundle
sonata-project/core-bundle               3.18.0 3.18.0 Symfony SonataCoreBundle (abandoned)
sonata-project/datagrid-bundle           2.5.0  3.1.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.6.0  1.6.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.15.0 3.15.0 Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  2.2.0  2.2.0  Lightweight Exporter library
sonata-project/intl-bundle               2.7.0  2.7.0  Symfony SonataIntlBundle
sonata-project/media-bundle              3.24.0 3.24.0 Symfony SonataMediaBundle
sonata-project/notification-bundle       3.6.2  3.6.2  Symfony SonataNotificationBundle
sonata-project/translation-bundle        2.4.2  2.4.2  SonataTranslationBundle
sonata-project/user-bundle               4.5.2  4.5.2  Symfony SonataUserBundle
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
Restricting packages listed in "symfony/symfony" to "4.4.*"
symfony/asset                      v4.4.5  v4.4.5  Symfony Asset Component
symfony/browser-kit                v4.4.5  v4.4.5  Symfony BrowserKit Component
symfony/cache                      v4.4.5  v4.4.5  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts            v2.0.1  v2.0.1  Generic abstractions related to caching
symfony/config                     v4.4.5  v4.4.5  Symfony Config Component
symfony/console                    v4.4.5  v4.4.5  Symfony Console Component
symfony/css-selector               v4.4.5  v4.4.5  Symfony CssSelector Component
symfony/debug                      v4.4.5  v4.4.5  Symfony Debug Component
symfony/debug-bundle               v4.4.5  v4.4.5  Symfony DebugBundle
symfony/debug-pack                 v1.0.7  v1.0.7  A debug pack for Symfony projects
symfony/dependency-injection       v4.4.5  v4.4.5  Symfony DependencyInjection Component
symfony/doctrine-bridge            v4.4.5  v4.4.5  Symfony Doctrine Bridge
symfony/dom-crawler                v4.4.5  v4.4.5  Symfony DomCrawler Component
symfony/dotenv                     v4.4.5  v4.4.5  Registers environment variables from a .env file
symfony/error-handler              v4.4.5  v4.4.5  Symfony ErrorHandler Component
symfony/event-dispatcher           v4.4.5  v4.4.5  Symfony EventDispatcher Component
symfony/event-dispatcher-contracts v1.1.7  v2.0.1  Generic abstractions related to dispatching event
symfony/expression-language        v4.4.5  v4.4.5  Symfony ExpressionLanguage Component
symfony/filesystem                 v4.4.5  v4.4.5  Symfony Filesystem Component
symfony/finder                     v4.4.5  v4.4.5  Symfony Finder Component
symfony/flex                       v1.6.2  v1.6.2  Composer plugin for Symfony
symfony/form                       v4.4.5  v4.4.5  Symfony Form Component
symfony/framework-bundle           v4.4.5  v4.4.5  Symfony FrameworkBundle
symfony/http-foundation            v4.4.5  v4.4.5  Symfony HttpFoundation Component
symfony/http-kernel                v4.4.5  v4.4.5  Symfony HttpKernel Component
symfony/inflector                  v4.4.5  v4.4.5  Symfony Inflector Component
symfony/intl                       v4.4.5  v4.4.5  A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/maker-bundle               v1.14.6 v1.14.6 Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writing boilerplate code.
symfony/mime                       v4.4.5  v4.4.5  A library to manipulate MIME messages
symfony/monolog-bridge             v4.4.5  v4.4.5  Symfony Monolog Bridge
symfony/monolog-bundle             v3.5.0  v3.5.0  Symfony MonologBundle
symfony/options-resolver           v4.4.5  v4.4.5  Symfony OptionsResolver Component
symfony/orm-pack                   v1.0.8  v1.0.8  A pack for the Doctrine ORM
symfony/phpunit-bridge             v5.0.5  v5.0.5  Symfony PHPUnit Bridge
symfony/polyfill-intl-icu          v1.14.0 v1.14.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.14.0 v1.14.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring          v1.14.0 v1.14.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.14.0 v1.14.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.14.0 v1.14.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/process                    v4.4.5  v4.4.5  Symfony Process Component
symfony/profiler-pack              v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access            v4.4.5  v4.4.5  Symfony PropertyAccess Component
symfony/property-info              v4.4.5  v4.4.5  Symfony Property Info Component
symfony/routing                    v4.4.5  v4.4.5  Symfony Routing Component
symfony/security-acl               v3.0.4  v3.0.4  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.5  v4.4.5  Symfony SecurityBundle
symfony/security-core              v4.4.5  v4.4.5  Symfony Security Component - Core Library
symfony/security-csrf              v4.4.5  v4.4.5  Symfony Security Component - CSRF Library
symfony/security-guard             v4.4.5  v4.4.5  Symfony Security Component - Guard
symfony/security-http              v4.4.5  v4.4.5  Symfony Security Component - HTTP Integration
symfony/serializer                 v4.4.5  v4.4.5  Symfony Serializer Component
symfony/serializer-pack            v1.0.2  v1.0.2  A pack for the Symfony serializer
symfony/service-contracts          v2.0.1  v2.0.1  Generic abstractions related to writing services
symfony/stopwatch                  v4.4.5  v4.4.5  Symfony Stopwatch Component
symfony/swiftmailer-bundle         v3.4.0  v3.4.0  Symfony SwiftmailerBundle
symfony/templating                 v4.4.5  v4.4.5  Symfony Templating Component
symfony/test-pack                  v1.0.6  v1.0.6  A pack for functional and end-to-end testing within a Symfony app
symfony/translation                v4.4.5  v4.4.5  Symfony Translation Component
symfony/translation-contracts      v2.0.1  v2.0.1  Generic abstractions related to translation
symfony/twig-bridge                v4.4.5  v4.4.5  Symfony Twig Bridge
symfony/twig-bundle                v4.4.5  v4.4.5  Symfony TwigBundle
symfony/validator                  v4.4.5  v4.4.5  Symfony Validator Component
symfony/var-dumper                 v4.4.5  v4.4.5  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter               v4.4.5  v4.4.5  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-link                   v4.4.5  v4.4.5  Symfony WebLink Component
symfony/web-profiler-bundle        v4.4.5  v4.4.5  Symfony WebProfilerBundle
symfony/web-server-bundle          v4.4.5  v4.4.5  Symfony WebServerBundle
symfony/yaml                       v4.4.5  v4.4.5  Symfony Yaml Component
```

#### PHP version

```
$ php -v
PHP 7.2.28-3+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Feb 23 2020 07:23:01) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.28-3+ubuntu16.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.9.2, Copyright (c) 2002-2020, by Derick Rethans
```

## Subject

`TypeError` in `DoctrineBackendOptimizeListener` after update from SF4.3 to SF4.4

## Steps to reproduce

Simply run and browse project. The error also occurs using command line at first iteration:
`bin/console sonata:notification:start --env=dev --iteration=250`

## Expected results

No error.

## Actual results

Everything seems to work well, but the following error is displayed under each page:
```
TypeError:
Argument 1 passed to Sonata\NotificationBundle\Event\DoctrineBackendOptimizeListener::__construct() must be an instance of Symfony\Bridge\Doctrine\RegistryInterface, instance of Doctrine\Bundle\DoctrineBundle\Registry given, called in /home/jo66/www/bizbox/var/cache/dev/ContainerTSbUnOZ/getSonata_Notification_Event_DoctrineBackendOptimizeService.php on line 12

  at vendor/sonata-project/notification-bundle/src/Event/DoctrineBackendOptimizeListener.php:32
  at Sonata\NotificationBundle\Event\DoctrineBackendOptimizeListener->__construct(object(Registry))
     (var/cache/dev/ContainerTSbUnOZ/getSonata_Notification_Event_DoctrineBackendOptimizeService.php:12)
  at require('/home/jo66/www/bizbox/var/cache/dev/ContainerTSbUnOZ/getSonata_Notification_Event_DoctrineBackendOptimizeService.php')
     (var/cache/dev/ContainerTSbUnOZ/srcApp_KernelDevDebugContainer.php:857)
  at ContainerTSbUnOZ\srcApp_KernelDevDebugContainer->load('getSonata_Notification_Event_DoctrineBackendOptimizeService.php')
     (var/cache/dev/ContainerTSbUnOZ/srcApp_KernelDevDebugContainer.php:1096)
  at ContainerTSbUnOZ\srcApp_KernelDevDebugContainer->ContainerTSbUnOZ\{closure}()
     (vendor/symfony/event-dispatcher/EventDispatcher.php:279)
  at Symfony\Component\EventDispatcher\EventDispatcher->sortListeners('sonata.notification.event.message_iterate_event')
     (vendor/symfony/event-dispatcher/EventDispatcher.php:98)
  at Symfony\Component\EventDispatcher\EventDispatcher->getListeners('sonata.notification.event.message_iterate_event')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:102)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->getListeners()
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:216)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->getNotCalledListeners(null)
     (vendor/symfony/http-kernel/DataCollector/EventDataCollector.php:69)
  at Symfony\Component\HttpKernel\DataCollector\EventDataCollector->lateCollect()
     (vendor/symfony/http-kernel/Profiler/Profiler.php:100)
  at Symfony\Component\HttpKernel\Profiler\Profiler->saveProfile(object(Profile))
     (vendor/symfony/http-kernel/EventListener/ProfilerListener.php:112)
  at Symfony\Component\HttpKernel\EventListener\ProfilerListener->onKernelTerminate(object(TerminateEvent), 'kernel.terminate', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:126)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(TerminateEvent), 'kernel.terminate', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:264)
  at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.terminate', object(TerminateEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:239)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.terminate', object(TerminateEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:73)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(TerminateEvent), 'kernel.terminate')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:168)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(TerminateEvent), 'kernel.terminate')
     (vendor/symfony/http-kernel/HttpKernel.php:88)
  at Symfony\Component\HttpKernel\HttpKernel->terminate(object(Request), object(Response))
     (vendor/symfony/http-kernel/Kernel.php:166)
  at Symfony\Component\HttpKernel\Kernel->terminate(object(Request), object(Response))
     (public/index.php:27)
```

**Comment by jo66** [comment]
Changing the type of argument in `DoctrineBackendOptimizeListener`'s constructor from `RegistryInterface` to `Registry` solve the issue, but I didn't want to make a PR because maybe the problem is linked with version of my packages.

**Comment by jo66** [comment]
SF5 packages appear in environment section, but it happened when I updated from SF4.3 to SF4.4, with the following packages installed:
```
sonata-project/admin-bundle              3.63.0 3.63.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.18.4 3.18.4 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/classification-bundle     3.11.1 3.11.1 Symfony SonataClassificationBundle
sonata-project/core-bundle               3.18.0 3.18.0 Symfony SonataCoreBundle (abandoned)
sonata-project/datagrid-bundle           2.5.0  3.1.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.6.0  1.6.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.15.0 3.15.0 Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  2.2.0  2.2.0  Lightweight Exporter library
sonata-project/intl-bundle               2.7.0  2.7.0  Symfony SonataIntlBundle
sonata-project/media-bundle              3.24.0 3.24.0 Symfony SonataMediaBundle
sonata-project/notification-bundle       3.6.2  3.6.2  Symfony SonataNotificationBundle
sonata-project/translation-bundle        2.4.2  2.4.2  SonataTranslationBundle
sonata-project/user-bundle               4.5.2  4.5.2  Symfony SonataUserBundle
```
```
Restricting packages listed in "symfony/symfony" to "4.4.*"
symfony/asset                      v4.4.5  v4.4.5  Symfony Asset Component
symfony/browser-kit                v4.4.5  v4.4.5  Symfony BrowserKit Component
symfony/cache                      v4.4.5  v4.4.5  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts            v2.0.1  v2.0.1  Generic abstractions related to caching
symfony/config                     v4.4.5  v4.4.5  Symfony Config Component
symfony/console                    v4.4.5  v4.4.5  Symfony Console Component
symfony/css-selector               v4.4.5  v4.4.5  Symfony CssSelector Component
symfony/debug                      v4.4.5  v4.4.5  Symfony Debug Component
symfony/debug-bundle               v4.4.5  v4.4.5  Symfony DebugBundle
symfony/debug-pack                 v1.0.7  v1.0.7  A debug pack for Symfony projects
symfony/dependency-injection       v4.4.5  v4.4.5  Symfony DependencyInjection Component
symfony/doctrine-bridge            v4.4.5  v4.4.5  Symfony Doctrine Bridge
symfony/dom-crawler                v4.4.5  v4.4.5  Symfony DomCrawler Component
symfony/dotenv                     v4.4.5  v4.4.5  Registers environment variables from a .env file
symfony/error-handler              v4.4.5  v4.4.5  Symfony ErrorHandler Component
symfony/event-dispatcher           v4.4.5  v4.4.5  Symfony EventDispatcher Component
symfony/event-dispatcher-contracts v1.1.7  v2.0.1  Generic abstractions related to dispatching event
symfony/expression-language        v4.4.5  v4.4.5  Symfony ExpressionLanguage Component
symfony/filesystem                 v4.4.5  v4.4.5  Symfony Filesystem Component
symfony/finder                     v4.4.5  v4.4.5  Symfony Finder Component
symfony/flex                       v1.6.2  v1.6.2  Composer plugin for Symfony
symfony/form                       v4.4.5  v4.4.5  Symfony Form Component
symfony/framework-bundle           v4.4.5  v4.4.5  Symfony FrameworkBundle
symfony/http-foundation            v4.4.5  v4.4.5  Symfony HttpFoundation Component
symfony/http-kernel                v4.4.5  v4.4.5  Symfony HttpKernel Component
symfony/inflector                  v4.4.5  v4.4.5  Symfony Inflector Component
symfony/intl                       v4.4.5  v4.4.5  A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/maker-bundle               v1.14.6 v1.14.6 Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writing boilerplate code.
symfony/mime                       v4.4.5  v4.4.5  A library to manipulate MIME messages
symfony/monolog-bridge             v4.4.5  v4.4.5  Symfony Monolog Bridge
symfony/monolog-bundle             v3.5.0  v3.5.0  Symfony MonologBundle
symfony/options-resolver           v4.4.5  v4.4.5  Symfony OptionsResolver Component
symfony/orm-pack                   v1.0.8  v1.0.8  A pack for the Doctrine ORM
symfony/phpunit-bridge             v5.0.5  v5.0.5  Symfony PHPUnit Bridge
symfony/polyfill-intl-icu          v1.14.0 v1.14.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.14.0 v1.14.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring          v1.14.0 v1.14.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.14.0 v1.14.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.14.0 v1.14.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/process                    v4.4.5  v4.4.5  Symfony Process Component
symfony/profiler-pack              v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access            v4.4.5  v4.4.5  Symfony PropertyAccess Component
symfony/property-info              v4.4.5  v4.4.5  Symfony Property Info Component
symfony/routing                    v4.4.5  v4.4.5  Symfony Routing Component
symfony/security-acl               v3.0.4  v3.0.4  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.5  v4.4.5  Symfony SecurityBundle
symfony/security-core              v4.4.5  v4.4.5  Symfony Security Component - Core Library
symfony/security-csrf              v4.4.5  v4.4.5  Symfony Security Component - CSRF Library
symfony/security-guard             v4.4.5  v4.4.5  Symfony Security Component - Guard
symfony/security-http              v4.4.5  v4.4.5  Symfony Security Component - HTTP Integration
symfony/serializer                 v4.4.5  v4.4.5  Symfony Serializer Component
symfony/serializer-pack            v1.0.2  v1.0.2  A pack for the Symfony serializer
symfony/service-contracts          v2.0.1  v2.0.1  Generic abstractions related to writing services
symfony/stopwatch                  v4.4.5  v4.4.5  Symfony Stopwatch Component
symfony/swiftmailer-bundle         v3.4.0  v3.4.0  Symfony SwiftmailerBundle
symfony/templating                 v4.4.5  v4.4.5  Symfony Templating Component
symfony/test-pack                  v1.0.6  v1.0.6  A pack for functional and end-to-end testing within a Symfony app
symfony/translation                v4.4.5  v4.4.5  Symfony Translation Component
symfony/translation-contracts      v2.0.1  v2.0.1  Generic abstractions related to translation
symfony/twig-bridge                v4.4.5  v4.4.5  Symfony Twig Bridge
symfony/twig-bundle                v4.4.5  v4.4.5  Symfony TwigBundle
symfony/validator                  v4.4.5  v4.4.5  Symfony Validator Component
symfony/var-dumper                 v4.4.5  v4.4.5  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter               v4.4.5  v4.4.5  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-link                   v4.4.5  v4.4.5  Symfony WebLink Component
symfony/web-profiler-bundle        v4.4.5  v4.4.5  Symfony WebProfilerBundle
symfony/web-server-bundle          v4.4.5  v4.4.5  Symfony WebServerBundle
symfony/yaml                       v4.4.5  v4.4.5  Symfony Yaml Component
```
I should go back to 4.4 before opening issue, sorry.

**Comment by jo66** [comment]
Fixed by b65e976
